pkg/liqo-controller-manager/webhooks/foreigncluster/fc.go

```diff
@@ -25,15 +26,28 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
 
 	discoveryv1alpha1 "github.com/liqotech/liqo/apis/discovery/v1alpha1"
+	"github.com/liqotech/liqo/pkg/discovery"
 )
 
 type fcwh struct {
 	decoder *admission.Decoder
 }
 
-// New returns a new NamespaceOffloadingWebhook instance.
-func New() *webhook.Admission {
-	return &webhook.Admission{Handler: &fcwh{}}
+type fcwhm struct {
+	fcwh
+}
+type fcwhv struct {
+	fcwh
+}
+
+// NewMutate returns a new ForeignCluster mutating webhook.
+func NewMutate() *webhook.Admission {
```

Comment:
```suggestion
func NewMutator() *webhook.Admission {
```

Seems slightly better to me. The same for Validator.